Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,319 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crypto
+import Foundation
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+
+/// Types associated with the ML-DSA-65 algorithm
+@_documentation(visibility: public)
+public enum MLDSA {}
+
+extension MLDSA {
+    /// A ML-DSA-65 private key.
+    public struct PrivateKey: Sendable {
+        fileprivate let pointer: UnsafeMutablePointer<MLDSA65_private_key>
```

Comment:
This is unsound. It's never possible for a copyable `struct` to own an unsafe pointer safely. This needs to have a backing class to own the pointer.